espnet2/tasks/hubert.py

```diff
@@ -282,8 +283,37 @@ def build_collate_fn(
             pad=args.collate_fn_conf.get("pad", False),
             rand_crop=args.collate_fn_conf.get("rand_crop", True),
             crop_audio=not args.collect_stats,
+            kernel_size=args.collate_fn_conf["kernel_size"],
+            stride=args.collate_fn_conf["stride"],
+            sample_rate=args.collate_fn_conf["sample_rate"],
         )
 
+    @classmethod
+    def update_collate_info(cls, args: argparse.Namespace) -> None:
+        assert check_argument_types()
+
+        fs = args.frontend_conf.get("fs", None)
+        # to conduct collate, fs must be included in frontend_conf
+        assert fs is not None, "sample rate is necessary to update collate fn"
+        if isinstance(fs, str):
+            fs = humanfriendly.parse_size(fs)
+        sample_rate = fs / 1000
+
+        if args.encoder_conf.get("extractor_conv_layer_config", None) is None:
+            # corresponding to default conv extractor
+            # refer to espnet2/asr/encoder/hubert_encoder.py
+            reception_field = 400
+            stride_field = 320
+        else:
+            stride_field, reception_field = 1, 1
+            for conv_config in args.encoder_conf["extractor_conv_layer_config"][::-1]:
+                _, kernel, stride = conv_config
+                stride_field *= stride
+                reception_field = stride * (reception_field - 1) + kernel
+        args.collate_fn_conf["kernel_size"] = reception_field / sample_rate
```

Comment:
kernel_size / stride are a bit ambiguous with convolution terms. Is there a tradition or special reason to use them?

Reply:
I'm following your original naming as it shows in the collate_fn. Frankly, I do not know if there is a naming convention to this. I can change it